app/api/discount_codes.py

```diff
@@ -23,14 +23,27 @@ class Meta:
         self_view_kwargs = {'id': '<id>'}
         inflect = dasherize
 
-    @validates_schema
-    def validate_quantity(self, data):
-        if 'max_quantity' in data and 'min_quantity' in data:
-            if data['max_quantity'] < data['min_quantity']:
+    @validates_schema(pass_original=True)
+    def validate_quantity(self, data, original_data):
+        original_data_attributes = original_data['data']['attributes']
+        discount_code = DiscountCode.query.filter_by(id=original_data['data']['id']).one()
+        if 'min_quantity' in original_data_attributes:
```

Comment:
Just wondering why you are checking in original_data_attributes. I know it doesn't makes any difference in output/login but it makes sense to check on validated data object i.e, `data` here. 
`data` here is already processed with initial validations. Please correct me If I'm wrong here @LuD1161 @shubham-padia